scripts/globals/dark_ixion.lua

```diff
@@ -462,3 +584,292 @@ xi.darkixion.onMobDisengage = function(mob)
+                    local stomp = false
+
+                    local dork = nearbyPlayers[i]
+                    if dork:isAlive() then
+                        local nextHit = dork:getID()
+
+                        if #xi.darkixion.hitList == 0 then
+                            table.insert(xi.darkixion.hitList, nextHit)
+                            stomp = true
+                        else
+                            stomp = true
+                            for v = 1, #xi.darkixion.hitList do
+                                if xi.darkixion.hitList[v] == nextHit then
+                                    stomp = false
+                                end
+                            end
+
+                            if stomp then
+                                table.insert(xi.darkixion.hitList, nextHit)
+                                stomp = true
+                            end
+                        end
+
+                        -- if dork hasn't been trampled yet in this path
+                        -- or if dork is the original target of the trample (you better run away)
+                        if
+                            stomp and
+                            (dork:isInfront(mob, 30) or
+                            dork:getTargID() == mob:getLocalVar('trampleTargID'))
+                        then
+                            mob:useMobAbility(2339, dork) -- trample
+                        end
+                    end
+                end
+            end
+        else
+            xi.darkixion.endStomp(mob)
+        end
+    end
+end
+
+xi.darkixion.endStomp = function(mob)
+    mob:setLocalVar('trampling', 0)
+    mob:setAnimationSub(mob:getLocalVar('originalSub'))
+    mob:setSpeed(70)
+    mob:setAutoAttackEnabled(true)
+    mob:setMobAbilityEnabled(true)
+    mob:setLocalVar('lastHit', 0)
+    mob:setLocalVar('trampleCount', mob:getLocalVar('trampleCount') - 1)
+
+    -- Don't move around if doing multiple tramples
+    if mob:getLocalVar('trampleCount') > 0 then
+        mob:setLocalVar('timeSinceWS', os.time() + 2)
+    end
+
+    mob:timer(2000, function(mobArg)
+        if mobArg:getLocalVar('trampling') == 0 then
+            mobArg:setBehaviour(0)
+        end
+    end)
+
+    mob:setMobSkillAttack(39)
+
+    xi.darkixion.pos     = nil
+    xi.darkixion.hitList = nil
+end
+
+xi.darkixion.itsStompinTime = function(mob, target)
+    local targets = {}
+    -- hitList is a list of all players considered for this trample. If they get out of the way, they do not get hit
+    xi.darkixion.hitList = {}
+
+    mob:setLocalVar('trampling', 1)
+
+    -- remove bind
+    mob:delStatusEffect(xi.effect.BIND)
+    mob:delStatusEffect(xi.effect.WEIGHT)
+    mob:setSpeed(95)
+    mob:setAutoAttackEnabled(false)
+    mob:setMobAbilityEnabled(false)
+    mob:setMobSkillAttack(0)
+    mob:setBehaviour(xi.behavior.NO_TURN + xi.behavior.STANDBACK)
+
+    local nearbyPlayers = mob:getPlayersInRange(30)
```

Comment:
So does this include ALL players, even those not in the fighting alliance or on the enmity list?

Reply:
yes. can even hit people on a chocobo

Reply:
kinda late for me to mention this, bit Nidhog's spike flail is the same way - he'll whip your rear for being in the vicinity. I forget whether that was handled with a target flag or a skill flag. Had this been a skill script instead of a mob script, it'd get handled for you.

Reply:
In this case we aren't actually triggering an ability, we're choosing a target to run to then every tick looking for nearby targets to see if they get trampled

Reply:
I know, I wasn't confused. its unfortunate the only place we have a pre-made handler of it is in mobskills.

Reply:
I believe Zach said he'd implement something for lua to arbitrarily use target find